Sources/PitchSpeller/Data Structures/Pair.swift

```diff
@@ -25,8 +25,39 @@ public struct Pair<A, B, Orderedness> {
 extension Pair: Hashable where A: Hashable, B: Hashable { }
 extension Pair: Equatable where A: Equatable, B: Equatable { }
 
-extension Pair where Orderedness == Ordered {
+extension Pair {
     public var reversed: Pair<B, A, Orderedness> {
         return Pair<B, A, Orderedness>(b, a)
     }
 }
+
+/// map for elements of Pair
+extension Pair {
+    public func map<T, U> (_ aTransform: (A) -> T, _ bTransform: (B) -> U) -> Pair <T, U, Orderedness> {
```

Comment:
Super nit: I've taken on this style for generics. I could easily be convinced away from this, but so far I like it.

Declaring generic methods, types, etc:

```Swift
func map <T,U> (...) { ... }
func == <T: A, U: B> (...) { ... }
struct Struct <T: A, U: B> { ... }
```

Referencing generic types:

```Swift
func map <T,U> (...) -> Pair<T,U,Orderedness>
```

Synopsis: extra whitespace padding when declaring (because that's where you might constrain the generic types to a constellation of protocols), no whitespace when referencing (because you will not be adding any constraints, and the commas do the job for delimiting).